psycop/common/cohort_definition.py

```diff
@@ -1,20 +1,18 @@
 from abc import ABC, abstractmethod
 from collections.abc import Iterable
+from typing import Protocol, runtime_checkable
 
 import polars as pl
+from wasabi import Printer
 
 from psycop.common.global_utils.pydantic_basemodel import PSYCOPBaseModel
 
 
-class PredictionTimeFilter(ABC):
-    """Interface for filtering functions applied to prediction times. Requried
-    to have an `apply` method that takes a `pl.DataFrame` (the unfiltered
-    prediction times) and returns a `pl.DataFrame` (the filtered prediction times).
-    """
+@runtime_checkable
+class PredictionTimeFilter(Protocol):
```

Comment:
just realised that we have a PredictionTimeFilter and a PredictionTimeFilter**er** class 🙃